### PR TITLE
Electrum Wallet doc overhaul with BTCPay / ElectrumX integration docs

### PR DESCRIPTION
**Pull request by maltokyo**

### CODE REVIEW

**Comment by maltokyo** [comment]
Thanks a lot for all the valid feedback. Really nice. Indeed was my first
time to write something like this, normally I’m hacking stuff to work by
myself but never had to take the time to explain it to others, I learn a
lot from all of you guys.

File: ElectrumWallet.md [hunk 1]
```diff
@@ -100,3 +106,96 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+But, let's say you DON'T want to rely on or trust anyone else with your detailed blockchain queries and transactions.  After all, sending the entirety of your online wallet activity (including your IP address, receive/send addresses, amounts etc) through another server as your "proxy" to the blockchain might give you the heebie-jeebies.  If this is you, read on!
+
+## Section 2.1 Enable Your Own ElectrumX Server (fully integrated with your BTCPay Server's full bitcoin node)
+
+### Prerequisites (mandatory):
+
+1. You are using or will use the [Docker version of BTCPay Server](https://github.com/btcpayserver/btcpayserver-docker)
+2. You do NOT [prune your full bitcoin node](https://docs.btcpayserver.org/faq-and-common-issues/faq-synchronization#can-i-skip-the-synchronization) in BTCPay Server (i.e. you have synched and stored from genesis block, and you do not use the opt-save-storage [Environment Variable](https://docs.btcpayserver.org/faq-and-common-issues/faq-deployment#how-can-i-modify-deactivate-environment-variables))
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Done

File: ElectrumWallet.md [hunk 2]
```diff
@@ -100,3 +106,96 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+![ElectrumWalletServerList](https://user-images.githubusercontent.com/1388507/68437521-8a5eb580-01c1-11ea-9ece-0666353a6742.png)
+
+While there is nothing wrong with "Select Server Automatically" per se, every transaction that you make/browse/broadcast in your Electrum Wallet will be using someone else's server to do it all.  Don't take this point as a threat, that is exactly what the Electrum server network is there to do and indeed how most users will safely use their wallet.  The people running those ElectrumX public servers are just like us, bitcoin enthusiasts who want to help adoption by playing their part.  If you are not (yet) into becoming your own completely sovereign and private bitcoin "bank" there is no need for you to read further, because everything is hunky-dory.
+
+But, let's say you DON'T want to rely on or trust anyone else with your detailed blockchain queries and transactions.  After all, sending the entirety of your online wallet activity (including your IP address, receive/send addresses, amounts etc) through another server as your "proxy" to the blockchain might give you the heebie-jeebies.  If this is you, read on!
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Done

File: ElectrumWallet.md [hunk 3]
```diff
@@ -100,3 +106,96 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+2. You do NOT [prune your full bitcoin node](https://docs.btcpayserver.org/faq-and-common-issues/faq-synchronization#can-i-skip-the-synchronization) in BTCPay Server (i.e. you have synched and stored from genesis block, and you do not use the opt-save-storage [Environment Variable](https://docs.btcpayserver.org/faq-and-common-issues/faq-deployment#how-can-i-modify-deactivate-environment-variables))
+3. You have at least 400GB of drive space on the drive where your docker volumes are stored (as at the writing of this documentation on 8th Nov 2019, my BTCPay Server volumes directory uses in total is 333G with full node and ElectrumX enabled and obviously it will grow further with time).
+4. You are familiar with how to use BTCPays [Additional Fragment](https://github.com/btcpayserver/btcpayserver-docker/blob/master/README.md#environment-variables) feature as part of your environment variable setup.
+5. Basic Linux command line knowledge: I guess you have that already, as you are reading this far.  This is tested fully on Ubuntu 18.04 but we assume this will work on any linux with BTCPay server and docker.
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Done

File: ElectrumWallet.md [hunk 4]
```diff
@@ -100,3 +106,96 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+## Section 2.1 Enable Your Own ElectrumX Server (fully integrated with your BTCPay Server's full bitcoin node)
+
+### Prerequisites (mandatory):
+
+1. You are using or will use the [Docker version of BTCPay Server](https://github.com/btcpayserver/btcpayserver-docker)
+2. You do NOT [prune your full bitcoin node](https://docs.btcpayserver.org/faq-and-common-issues/faq-synchronization#can-i-skip-the-synchronization) in BTCPay Server (i.e. you have synched and stored from genesis block, and you do not use the opt-save-storage [Environment Variable](https://docs.btcpayserver.org/faq-and-common-issues/faq-deployment#how-can-i-modify-deactivate-environment-variables))
+3. You have at least 400GB of drive space on the drive where your docker volumes are stored (as at the writing of this documentation on 8th Nov 2019, my BTCPay Server volumes directory uses in total is 333G with full node and ElectrumX enabled and obviously it will grow further with time).
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Fixed

File: ElectrumWallet.md [hunk 5]
```diff
@@ -100,3 +106,96 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+If you have been running your BTCPay Server for a while but haven't had `txindex=1` set until now, then it might take a few hours to build the index, this is no issue and it should not involve downtime of more than a few hours - better to set this to run overnight though when nobody will be using your node. Note: If you want to rebuild the index from scratch, launch bitcoind once with the `reindex=1` option (warning: this reindex option may take a VERY long time, and is not enabled out of the box as you likely dont need it, and hence is not in scope of this document).
+
+### OK OK, I got it, now what do I have to do to make it work?!:
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Fixed

File: ElectrumWallet.md [hunk 6]
```diff
@@ -100,3 +106,96 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+If you have read this far (you're a star!) and theres not much to do in order to set it all up.  Below are the steps to enable it in your BTCPay node.
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Removed

File: ElectrumWallet.md [hunk 7]
```diff
@@ -100,3 +106,96 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+1. ElectrumX Server is accessible for Electrum Wallets via TCP port 50002.  You need to open this port up fully at least to be available within your own network to any PC or Android device running Electrum Wallet, and turn on port forwarding (you really should setup port forwarding, and open 50002 up on your router to the whole world so that your ElectrumX Server becomes part of the Electrum Server ecosystem and becomes available to other Electrum wallet users on the internet.  There is no risk to do this, port 50002 is SSL encrypted out of the box, and is designed to accept connections from unknown Electrum Users.  Although, it is assumed you have this knowledge already if you want to do that, as there are too many router models out there to cover it in the scope of this document).
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Shortened, with just the fact that you can do it, to be part of the ElectrumX network

**Comment by maltokyo** [comment]
Hi @pavlenex all edits are done, and I even added some further hard core privacy pro-tips.  Please review.

**Comment by pavlenex** [comment]
Awesome, thanks for quick fixes, please give me a day to review these and provide additional feedback if needed. Mike and Brit should provide some feedback soon as well 🚀

File: ElectrumWallet.md [hunk 8]
```diff
@@ -100,3 +106,113 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+If you have been running your BTCPay Server for a while but haven't had `txindex=1` set until now, then it might take a few hours to build the index, this is no issue and it should not involve downtime of more than a few hours - better to set this to run overnight though when nobody will be using your node. Note: If you want to rebuild the index from scratch, launch bitcoind once with the `reindex=1` option (warning: this reindex option may take a VERY long time, and is not enabled out of the box as you likely dont need it, and hence is not in scope of this document).
+
+### Steps to enable ElectrumX Server in BTCPay:
+
+Here are al the steps to enable ElectrumX Server in your BTCPay node (read carefully as you may need to adjust for your specific setup, especially if you use other custom or conflicting "fragments" (pruning, less-memory etc).
```
Comment (britttttk) [inline ElectrumWallet.md]: al

**Comment by britttttk** [comment]
#363 Awesome @maltokyo :)

File: ElectrumWallet.md [hunk 9]
```diff
@@ -100,3 +106,113 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+1. ElectrumX Server is accessible for Electrum Wallets via TCP port 50002.  You need to open this port up fully at least to be available within your own network to any PC or Android device running Electrum Wallet, and turn on port forwarding (you can also port forward 50002 from your Internet/WAN, to enable other Electrum Wallet users from the Internet to query your server).
+
+2. Enable the Docker Additional Fragment on your BTCPay node by running the following commands (this is assuming a brand new BTCPay installation with LND and ElectrumX, please tweak accordingly using the [relevant documentation](https://github.com/btcpayserver/btcpayserver-docker/blob/master/README.md#generated-docker-compose-):
+
+3. Follow the [normal setup and install of BTCPay Server](https://github.com/btcpayserver/btcpayserver-docker#full-installation-for-technical-users), then after this command `cd btcpayserver-docker`, follow the below instructions instead of those in the link.  If you already have a BTCPay Server running, then just follow from the next step.
```
[the remark this answers is not in the document]
Comment (maltokyo) [reply ElectrumWallet.md]: Hi @britttttk I have made this super clear now, please take a look.

File: ElectrumWallet.md (hunk 8 above)
Comment (maltokyo) [reply ElectrumWallet.md]: Hi @britttttk - thanks for your review, appreciated!  "al"?  Is it short for swipe right?

**Comment by maltokyo** [comment]
All done, please review final changes and merge PR if it is ok, as I already get several private messages on "how to" implement this.

File: ElectrumWallet.md (hunk 8 above)
Comment (britttttk) [reply ElectrumWallet.md]: Just a typo should be all, not a big deal :) thanks for adding the pruning warning.

File: ElectrumWallet.md [hunk 10]
```diff
@@ -100,3 +106,114 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+If you have been running your BTCPay Server for a while but haven't had `txindex=1` set until now, then it might take a few hours to build the index, this is no issue and it should not involve downtime of more than a few hours - better to set this to run overnight though when nobody will be using your node. Note: If you want to rebuild the index from scratch, launch bitcoind once with the `reindex=1` option (warning: this reindex option may take a VERY long time, and is not enabled out of the box as you likely dont need it, and hence is not in scope of this document).
+
+### Steps to enable ElectrumX Server in BTCPay:
+
+Here are al the steps to enable ElectrumX Server in your BTCPay node (read carefully as you may need to adjust for your specific setup, especially if you use other custom or conflicting "fragments" (pruning, less-memory etc.  To reiterate, ou should NOT proceed further here if you run a pruned BTCPay node.  In a [future release we will integrate EPS](https://github.com/btcpayserver/btcpayserver-docker/pull/227) into BTCPay, which will enable similar functionality as ElectrumX on a pruned node, but only for a personal wallet).
```
Comment (pavlenex) [inline ElectrumWallet.md]: > Here are al the steps

Typo - all

Comment (pavlenex) [inline ElectrumWallet.md]: >  To reiterate, ou should NOT 

Typo - you

File: ElectrumWallet.md [hunk 11]
```diff
@@ -100,3 +106,114 @@ There's no good answer to how much you should set the gap limit to.  Most mercha
...
+
+1. ElectrumX Server is accessible for Electrum Wallets via TCP port 50002.  You need to open this port up fully at least to be available within your own network to any PC or Android device running Electrum Wallet, and turn on port forwarding (you can also port forward 50002 from your Internet/WAN, to enable other Electrum Wallet users from the Internet to query your server).
```
Comment (pavlenex) [inline ElectrumWallet.md]: Just an idea, should we externally link somewhere how to open a port, or is that a bit too specific for this kind of doc? I assume newbies won't do this process, but still, this is just an idea, up to you. Maybe it'll be an overkill, whatever you decide.

**Comment by pavlenex** [comment]
ping @britttttk What's your take on [this](https://github.com/btcpayserver/btcpayserver-doc/pull/367#pullrequestreview-316244170)?

File: ElectrumWallet.md (hunk 11 above)
Comment (britttttk) [reply ElectrumWallet.md]: I would merge and open new issue for that.

**Comment by pavlenex** [comment]
@maltokyo Can you fix those typos so that we can merge this as is now, and later when EPS needs to be documented, we can split into two separate documents and make it a child of ElectrumWallet.md

**Comment by maltokyo** [comment]
done ✅ @pavlenex

**Comment by pavlenex** [comment]
@maltokyo I think you forgot to push the commits :)

**Comment by pavlenex** [comment]
Actually it's because of mine mess with trashing the repo, you're now on the wrong fork. Might be easier to fork and add a new pull request, I'm not aware of the other easy approach on this one, sorry my bad.
Screenshot 2019-11-18 at 09 42 35

**Comment by maltokyo** [comment]
Sorry, been a busy couple of weeks.  So, we delete this PR right, and start a new one?

**Comment by pavlenex** [comment]
@maltokyo yeah, let's close this one and just copy into appropriate fork.

**Comment by pavlenex** [comment]
#387

**Pull request closed without merging**